.devcontainer/vmr-source-build/build.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+set -ex
+
+source="${BASH_SOURCE[0]}"
+script_root="$( cd -P "$( dirname "$source" )" && pwd )"
+
+# TODO: Remove once we have a new version of the image
+dnf -y install openssl1.1
+
+installer_dir=$(realpath "$script_root/../..")
+workspace_dir=$(realpath "$installer_dir/../")
+tmp_dir=$(realpath "$workspace_dir/tmp")
+vmr_dir=$(realpath "$workspace_dir/dotnet")
+
+mkdir -p "$tmp_dir"
+
+# Codespaces performs a shallow fetch only
+git -C "$installer_dir" fetch --all --unshallow
```

Comment:
Why isn't shallow clone enough? Wouldn't folks who want to investigate another point in history rather just create the codespace from that commit?

Reply:
There are a few reasons for that but TLDR is that shallow clones have limited capabilities:
1. We need more commits than the one (like the one that is in the VMR currently) to be able to synchronize the VMR - to create a diff, we need at least two.
2. We fail to fetch from this shallow checkout into `tmp/installer` which we use during the sync - the temporary repo even fails to checkout a commit after the fetch as it only brings one shallow commit but not the parent.
3. We need to have history so that we can determine whether we want a `main` or `release/*` branch of the VMR (we need to know where the current PR branch is based from).